File: Backend/.NET/TicketManagementSystem_NET_Backend/TicketManagementSystem_NET_Backend/Repositories/UsersRepository/UsersRepository.cs
using Microsoft.Data.SqlClient;
using TicketManagementSystem_NET_Backend.Models;
using static Microsoft.EntityFrameworkCore.DbLoggerCategory.Database;

namespace TicketManagementSystem_NET_Backend.Repositories.UsersRepository
{
    public class UsersRepository : IUsersRepository
    {
        private readonly string connectionString = @"Data Source=ICHIM\SQLEXPRESS03;Initial Catalog=TicketManagementSystem;Persist Security Info=True;User ID=name;Password=password;TrustServerCertificate=True;encrypt=false;";
        public UsersRepository() { }

        public User Add(User t)
        {
            throw new NotImplementedException();
        }

        public Task<User> Delete(Guid id)
        {
            throw new NotImplementedException();
        }

        public async Task<bool> FindRoleByEmail(string email)
        {
            using (SqlConnection connection = new SqlConnection(connectionString))
            {
                await connection.OpenAsync();
                using (SqlCommand selectCommand = new SqlCommand(
                    "SELECT isCustomer FROM Users " +
                    "WHERE email = @email", connection))
                {
                    selectCommand.Parameters.AddWithValue("@email", email);
                    using (SqlDataReader reader = await selectCommand.ExecuteReaderAsync())
                    {
                        if (!reader.HasRows)
                        {
                            throw new Exception("User email not found");
                        }
                        reader.Read();
                        return reader.GetBoolean(0);
                    }
                }
            }
        }

        public IEnumerable<User> GetAll()
        {
            throw new NotImplementedException();
        }

        public Task<User> GetById(Guid id)
        {
            throw new NotImplementedException();
        }

        public Task<User> Update(User T)
        {
            throw new NotImplementedException();
        }
    }
}

File: Backend/Flask/data/UsersRepository.py
import pyodbc

class UsersRepository:
    def __init__(self):
        server = 'ICHIM\SQLEXPRESS03'
        database = 'TicketManagementSystem'
        username = ''
        password = ''
        self.connection_string = f'DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}'

    def find_role_by_email(self, email):
        with pyodbc.connect(self.connection_string) as connection:
            with connection.cursor() as cursor:
                print("enter")
                cursor.execute(
                    "SELECT isCustomer FROM Users WHERE email = ?", email)
                row = cursor.fetchone()
                if row is None:
                    raise Exception("User email not found")
                return row.isCustomer